Rapid7-ThreatCommand/ThreatCommand_workflow.xml

```diff
@@ -0,0 +1,57 @@
+<Workflow name="Qmasters Rapid7 ThreatCommand workflow for QRadar" version="1.0" xmlns="http://qradar.ibm.com/UniversalCloudRESTAPI/Workflow/V2">
+    <Parameters>
+		<Parameter name="hostname" label="Host Name" required="true" />
+		<Parameter name="account_id" label="Account ID" required="true" secret="true" />
+		<Parameter name="api_key" label="API Key" required="true" secret="true" />
+    </Parameters>
+	
+    <Actions>
+		<!-- Clear the log source status before a new workflow run starts -->
+        <ClearStatus />
+		
+        <!-- Prepare the start date filter, based on the current bookmark -->
+		<Initialize path="/bookmark" value="${time() - 604800000}" />
+        
+        <!-- Prepare the end date filter, which is the current time -->
+        <Set path="/current_time" value="${time()}" />
+
+		<!-- Get Alert List -->				
+		<CallEndpoint url="${/hostname}/public/v2/data/alerts/alerts-list" method="GET" savePath="/get_alert_list" >
+			<BasicAuthentication username="${/account_id}" password="${/api_key}" />
+            <QueryParameter name="isClosed" value="false" />
+            <QueryParameter name="foundDateFrom" value="${/bookmark}" />
+            <QueryParameter name="foundDateTo" value="${/current_time}" />
+		</CallEndpoint>
+
+        <!-- Handle Errors -->
+        <If condition="/get_alert_list/status_code = 403">
+			<Abort reason="Invalid ThreatCommand API Key"/>
+		</If>
+        <If condition="/get_alert_list/status_code != 200">
+            <Abort reason="hreatCommand abort reason:  ${/get_alert_list}" />
+        </If>
+
+        <!-- Get Alerts -->	
+        <ForEach item="/alert_id" items="/get_alert_list/body">
+		    <CallEndpoint url="${/hostname}/public/v1/data/alerts/get-complete-alert/${/alert_id}" method="GET" savePath="/get_alerts/${/alert_id}" >
+		    	<BasicAuthentication username="${/account_id}" password="${/api_key}" />
+		    </CallEndpoint>
+
+            <!-- Post Events, if any -->
+            <If condition="count(/get_alerts) > 0">
+                <PostEvent path="/get_alerts/${/alert_id}/body" source="${/hostname}" />

```

Comment:
```suggestion
	<!-- Initialize a list to contain all alerts -->
	<Set path="/alerts" value="[]" />

        <!-- Get Alerts -->	
        <ForEach item="/alert_id" items="/get_alert_list/body">
	    <CallEndpoint url="${/hostname}/public/v1/data/alerts/get-complete-alert/${/alert_id}" method="GET" savePath="/get_alert_${/alert_id}" >
	        <BasicAuthentication username="${/account_id}" password="${/api_key}" />
	    </CallEndpoint>
	    
	    <!-- Handle Errors -->
	    <If condition="/get_alert_${/alert_id}/status_code != 200">
	        <If condition="/get_alert_${/alert_id}/status_code = 404">
	             <Abort reason="Something something error with id something something"/>
	        </If>
	        
	        <Abort reason="ThreatCommand abort reason:  ${/get_alert_${/alert_id}}" />
	     </If>
	     
	     <Add path="/alerts" value=${/get_alert_${/alert_id}/body} />
	</ForEach>

	<!-- Post Events, if any -->
	<If condition="count(/alerts) > 0">
	     <PostEvent path="/alerts" source="${/hostname}" />
```
I've made some changes here:

1. Created a list that will hold every alert
2. Added an `if` statement that will check the validity of the response.
3. Changed the count count of the postevents to be on the created list.
4. posted the list.

The reason for these changes is to make as little `PostEvent` calls as possible.

**NOTE** - The changes I made are only under this scope and it might affect other areas in the code, so make so everything is unharmed.